sources/metricdog/src/host_check.rs

```diff
@@ -0,0 +1,15 @@
+use crate::error::Result;
+
+pub(crate) trait HostCheck {
+    /// Checks whether this current boot is the first boot of the host.
+    fn is_first_boot(&self) -> Result<bool>;
+
+    /// The time, in milliseconds, it took for the host to reach the 'preconfigured' stage.
+    fn preconfigured_time(&self) -> Result<String>;
+
+    /// The time, in milliseconds, it took for the host to reach the 'configured' stage.
+    fn configured_time(&self) -> Result<String>;
+
+    /// The time, in milliseconds, it took for the host network to become ready.
+    fn network_ready_time(&self) -> Result<String>;
```

Comment:
The time it took filesystems to become ready would also be interesting. I've found a couple boot time stalls there, and we end up doing more work in some scenarios (e.g. creating the data filesystem).

```suggestion
    /// The time, in milliseconds, it took for the host network to become ready.
    fn network_ready_time(&self) -> Result<String>;
    
    /// The time, in milliseconds, it took for the filesystems to become ready.
    fn filesystem_ready_time(&self) -> Result<String>;
```